vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -233,27 +233,43 @@ def apply(self,
               layer: torch.nn.Module,
               x: torch.Tensor,
               bias: Optional[torch.Tensor] = None) -> torch.Tensor:
+
         # ops.scaled_fp8_quant supports both dynamic and static quant.
         #   If dynamic, layer.act_scale is None and x_scale computed from x.
         #   If static,  layer.act_scale is scalar and x_scale set to act_scale.
-        qinput, x_scale = ops.scaled_fp8_quant(x,
-                                               layer.act_scale,
-                                               batch_dim_padding=17)
-
-        # Fused GEMM_DQ -- note we padded the input above because
-        # torch._scaled_mm is more performant for matrices with
-        # batch dimension > 16. Note that this could change
-        # in the future.
-        output, _ = torch._scaled_mm(
-            qinput,
-            layer.weight,
-            out_dtype=x.dtype,
-            scale_a=x_scale,
-            scale_b=layer.weight_scale,
-            bias=bias,
-        )
-
-        return torch.narrow(output, 0, 0, x.shape[0])
+
+        # We use the CUTLASS kernels by default but they don't support bias yet
+        if bias is None:
```

Comment:
Do we also do a branch if we are on ada lovelace and CUDA 12.1?

Reply:
We will need to if on CUDA < 12.4. We also need a branch if on CUDA 11.8. @comaniac do you know if `torch._scaled_mm` is supported in that case?

Reply:
I only know that it only supports SM89+. We can try to call this op with torch+cu118 to test out.

Reply:
The cutlass kernels need at least SM89 as well

Reply:
Yeah that makes sense. Older architectures don't have native FP8 so we can't get speedup from them, which seems not necessary to be covered.

Reply:
Note: we already have a mechanism for determining if a `LinearMethod` can run on a specific cuda arch. The `LinearMethod` exposes `get_min_capability` which is called during model loading.

https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/quantization/fp8.py#L46